### ISSUE TITLE
[ENH] Allow make_reduction to handle missing data

### ISSUE THREAD

**Issue by eenticott-shell** [sktime/sktime#4723]
Give the option of automatically handling missing data inside make_reduction. Currently if a model doesn't allow missing data (eg sklearn linear regression) then make_reduction will fail. 

Could set up so that for training data, rows with NA are removed and if NA appears in the required lags then those lags are removed with a warning. Should probably be an optional setting.

**Comment by fkiraly** [comment]
Can you expand why/how this is not sufficiently resolved by use of `Imputer`? I guess because you may want to have some rows in the inner `X` / `y` removed in this option?

**Comment by eenticott-shell** [comment]
Yes exactly, there may be cases where removing data/lags is preferable to imputation. Also for local pooling NAs can occur in the response which I think we want to avoid?

**Comment by fkiraly** [comment]
ah, I see. I've tried to deal with NAs in the `y` in the new prototype - hope to return to it in the summer.

Alternatively, happy for any PR that adds this functionality to the current classes!

**Comment by fkiraly** [comment]
FYI @danbartl

**Comment by eenticott-shell** [comment]
I have noticed an inconsistency between `local` and `global` pooling options that can cause problems for certain regressors. For example the `LGBMRegressor`. When fitting to a `y` containing missing data with `global` pooling the model fits without errors but `local` pooling throws an error due to NA in the response. I believe internally this is caused by the type of `Xt` that is called in `fit`. It is a pandas dataframe for `global` but a numpy array for `local`. If I have understood correctly then `local` and `global` should produce the same results when multindex data is not being used so it may be best to homogenise this.

```
# Imports
from sktime.forecasting.compose import make_reduction
from sktime.datasets import load_airline
import lightgbm as lgb

import numpy as np
import pandas as pd

# Load data
y = pd.DataFrame(load_airline())

def mask_y(y, n):
    mask_idx = np.random.choice(range(0, len(y)), n, replace = False)
    y.iloc[mask_idx] = pd.NA
    return(y)

# Mask random entries
y = mask_y(y, 15)

# Define regressor
regressor = lgb.LGBMRegressor()

# Try with global pooling
forecaster = make_reduction(estimator=regressor, 
                            strategy="recursive",
                            window_length=5,
                            transformers=None,
                            pooling='global')

forecaster.fit(y)

# Try with local pooling
forecaster = make_reduction(estimator=regressor, 
                            strategy="recursive",
                            window_length=5,
                            transformers=None,
                            pooling='local')

forecaster.fit(y)
```

**Comment by eenticott-shell** [comment]
Yes it definitely looks related, cleaning up the differing behaviours between pooling strategies would solve this too
